src/Lean/Meta/Tactic/Simp/Main.lean

```diff
@@ -19,6 +21,11 @@ register_builtin_option backward.dsimp.proofs : Bool := {
     descr    := "Let `dsimp` simplify proof terms"
   }
 
+register_builtin_option debug.simp.check : Bool := {
+    defValue := false
+    descr    := "enable consistency checks in simp"
```

Comment:
@kmill Right now, it performs only `have` consistency checks, correect?

Reply:
Yes; I'll rename this option to `debug.simp.check.have`. Later we can decide whether or not to have a `debug.simp.check` that enables consistency checks for all subsystems.